datadog-setup.php

```diff
@@ -221,6 +221,17 @@ function install($options)
         // Writing the ini file
         if ($phpProperties[INI_SCANDIR]) {
             $iniFileName = '98-ddtrace.ini';
+            // Search for pre-existing files with extension = ddtrace.so to avoid conflicts
+            // See issue https://github.com/DataDog/dd-trace-php/issues/1833
+            foreach (scandir($phpProperties[INI_SCANDIR]) as $ini) {
+                $path = "{$phpProperties[INI_SCANDIR]}/$ini";
+                if (is_file($path)) {
+                    if (preg_match('(^\s*extension\s*=.+ddtrace\.so)m', file_get_contents($path))) {
```

Comment:
.so suffix is not mandatory since 7.2, so pref may fail (and is different in other OS)

Reply:
Notice `=.+` (mandatory char) is also wrong, can be "extension=ddtrace" (no space), so need "=.*"  to swallow optional path
`preg_match("(^\s*extension\s*=.*ddtrace(\b|\.so))m"` seems OK.

Reply:
Perhaps, even:

```
if (preg_match("(^\s*extension\s*=\s*(\S*ddtrace)\b)m", file_get_contents($path), $res)
    && basename($res[1])=="ddtrace")  {
```